artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/buffer/TimedBuffer.java

```diff
@@ -34,6 +34,25 @@
 import org.apache.activemq.artemis.journal.ActiveMQJournalLogger;
 
 public final class TimedBuffer {
+
+   /**
+    * Property name to set the percentage of error allowed while expiring the flush {@code timeout} to happen:
+    * it can assume any positive value from {@code 0} to {@link Integer#MAX_VALUE}.
+    * <p>
+    * By default it is {@link #DEFAULT_TIMEOUT_ERROR_PERCENTAGE} more than the configured {@code timeout}.
+    */
+   public static final String JOURNAL_TIMEOUT_ERROR_PROPERTY_NAME = "journal.timeout.error";
```

Comment:
Why do we need another System property?

Reply:
I would remove the System.property here...
If this doesn't make sense in the configuration.. probably it's something users won't need to configure.

Reply:
You're right: TBH it was more for development purposes...
Next week I've planned to do some benchmarks with persistence enabled and I was thinking that would be nice to have it customizable :P
How it looks the rest of the refactored logic?